dex/farm/src/lib.rs

```diff
@@ -613,4 +613,214 @@ pub trait Farm:
             &context.get_input_attributes().unwrap().compounded_reward,
         )
     }
+
+    //This function is neccessary in old farms.
+    #[payable("*")]
+    #[endpoint(migrateToNewFarm)]
```

Comment:
This endpoint must be moved to v1_2 version of contracts. As those has to be updated first. We need to add test there.